web/src/main/java/org/fao/geonet/DatabaseMigration.java

```diff
@@ -166,19 +166,29 @@ boolean doMigration(String webappVersion, String subVersion, ServletContext serv
         // Migrate db if needed
         _logger.info("      Webapp   version:" + webappVersion + " subversion:" + subVersion);
         _logger.info("      Database version:" + dbVersion + " subversion:" + dbSubVersion);
-        if (dbVersion == null || webappVersion == null) {
-            _logger.warning("      Database does not contain any version information. Check that the database is a GeoNetwork "
-                + "database with data. The database is probably empty, no migration required.");
+        if (dbVersion == null){
+            _logger.warning("      Unable to retrieve the current GeoNetwork version from the database. "
+                + "If this is an initial run of the software, then the database will be auto-populated. "
+                + "Else check that the database is properly configured");
+            return true;
+        }
+        else if (webappVersion == null) {
```

Comment:
Please check the formatting: we have a mix of syntax in GeoNetwork, but I think is better to add the else if in the same line as the closing curly brace, like:

`} else if (webappVersion == null) {`